clusterctl/examples/google/provider-components.yaml.template

```diff
@@ -0,0 +1,354 @@
+          - name: config
+            mountPath: /etc/kubernetes
+          - name: certs
+            mountPath: /etc/ssl/certs
+        command:
+        - "./controller-manager"
+        args:
+        - --kubeconfig=/etc/kubernetes/admin.conf
+        resources:
+          requests:
+            cpu: 100m
+            memory: 20Mi
+          limits:
+            cpu: 100m
+            memory: 30Mi
+      - name: gce-machine-controller
+        image: gcr.io/k8s-cluster-api/gce-machine-controller:0.0.13
+        volumeMounts:
+          - name: config
+            mountPath: /etc/kubernetes
+          - name: certs
+            mountPath: /etc/ssl/certs
+          - name: credentials
+            mountPath: /etc/credentials
+          - name: sshkeys
+            mountPath: /etc/sshkeys
+          - name: machine-setup
+            mountPath: /etc/machinesetup
+          - name: kubeadm
+            mountPath: /usr/bin/kubeadm
+        env:
+          - name: GOOGLE_APPLICATION_CREDENTIALS
+            value: /etc/credentials/service-account.json
+        command:
+        - "./gce-machine-controller"
+        args:
+        - --kubeconfig=/etc/kubernetes/admin.conf
+        - --machinesetup=/etc/machinesetup/machine_setup_configs.yaml
+        resources:
+          requests:
+            cpu: 100m
+            memory: 20Mi
+          limits:
+            cpu: 100m
+            memory: 30Mi
+      volumes:
+      - name: config
+        hostPath:
+          path: /etc/kubernetes
+      - name: certs
+        hostPath:
+          path: /etc/ssl/certs
+      - name: sshkeys
+        secret:
+          secretName: machine-controller-sshkeys
+          defaultMode: 256
+      - name: credentials
+        secret:
+          secretName: machine-controller-credential
+      - name: machine-setup
+        configMap:
+          name: machine-setup
+      - name: kubeadm
+        hostPath:
+          path: /usr/bin/kubeadm
+---
+apiVersion: v1
+kind: Secret
+type: Opaque
+metadata:
+  name: machine-controller-sshkeys
+  namespace: default
+data:
+  user: $MACHINE_CONTROLLER_SSH_USER
+  private: $MACHINE_CONTROLLER_SSH_PRIVATE
+  # In-place master upgrade will not work till after https://github.com/kubernetes-sigs/cluster-api/issues/160
+  public: $MACHINE_CONTROLLER_SSH_PUBLIC
+---
+apiVersion: v1
+kind: Secret
+type: Opaque
+metadata:
+  name: machine-controller-credential
+  namespace: default
+data:
+  service-account.json: $MACHINE_CONTROLLER_SA_KEY
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: machine-setup
+data:
+  machine_setup_configs.yaml: |-
+    items:
+    - machineParams:
+      - os: ubuntu-1604-lts
+        roles:
+        - Master
+        versions:
+          kubelet: 1.9.4
+          controlPlane: 1.9.4
+          containerRuntime:
+            name: docker
+            version: 1.12.0
+      image: projects/ubuntu-os-cloud/global/images/family/ubuntu-1604-lts
+      metadata:
+        startupScript: |
+          set -e
```

Comment:
dumb question, is it possible to store the startup script in a *.sh file and then as part of the generate process insert it into this yaml?
If the startup script needs any modification, formatting it in a yaml seems very error-prone. at least when editing a sh file, editors/ide understands how to format it.

Reply:
I think this needs to be thought through more in the future. Right now, it matches the amount of encapsulation the configmap done in gcp-deployer. I am hesitant to introduce a lot of changes to pre-existing behavior (gcp-deployer) in a v0 PR.